packages/app-page-builder/src/blockEditor/components/elementSettingsTab/VariableSettings.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useMemo } from "react";
+import styled from "@emotion/styled";
+import { ButtonPrimary, ButtonSecondary } from "@webiny/ui/Button";
+import { PbEditorElement, PbBlockVariable } from "~/types";
+import { Form } from "@webiny/form";
+import { validation } from "@webiny/validation";
+import { Input } from "@webiny/ui/Input";
+import { useCurrentBlockElement } from "~/editor/hooks/useCurrentBlockElement";
+import { useConfirmationDialog } from "@webiny/app-admin/hooks/useConfirmationDialog";
+import { useUpdateElement } from "~/editor/hooks/useUpdateElement";
+
+const FormWrapper = styled("div")({
+    padding: "16px",
+    display: "grid",
+    rowGap: "16px"
+});
+
+const VariableSettings = ({ element }: { element: PbEditorElement }) => {
+    const { block } = useCurrentBlockElement();
+    const updateElement = useUpdateElement();
+
+    const initialData = useMemo(() => {
+        const variable = block?.data?.variables?.find(
+            (variable: PbBlockVariable) => variable.varRef === element?.data?.varRef
+        );
+
+        return { label: variable?.label };
+    }, [block, element]);
+
+    const onSubmit = useCallback(
+        formData => {
+            if (block && block.id) {
+                const newVariables = block.data?.variables?.map((variable: PbBlockVariable) => {
+                    if (variable?.varRef === element?.data?.varRef) {
```

Comment:
As I mentioned in the PR comment, let's use `variableId` within individual elements, instead of `varRef`.